mbuild/packing.py

```diff
@@ -127,30 +130,34 @@ def fill_box(compound, n_compounds=None, box=None, density=None, overlap=0.2,
                "must be specified. {} were given.".format(arg_count))
         raise ValueError(msg)
 
+    # Check other args
+    (compound, n_compounds, compound_ratio, fix_orientation,
+        compound_mass ) = _validate_packing_inputs(
+                                compound,
+                                n_compounds,
+                                compound_ratio,
+                                fix_orientation,
+                                compound_mass
+                            )
+
     if box is not None:
         box = _validate_box(box)
-    if not isinstance(compound, (list, set)):
-        compound = [compound]
-    if n_compounds is not None and not isinstance(n_compounds, (list, set)):
-        n_compounds = [n_compounds]
-    if not isinstance(fix_orientation, (list, set)):
-        fix_orientation = [fix_orientation]*len(compound)
-
-    if compound is not None and n_compounds is not None:
-        if len(compound) != len(n_compounds):
-            msg = ("`compound` and `n_compounds` must be of equal length.")
-            raise ValueError(msg)
-
-    if compound is not None:
-        if len(compound) != len(fix_orientation):
-            msg = ("`compound`, `n_compounds`, and `fix_orientation` "
-                   "must be of equal length.")
-            raise ValueError(msg)
 
     if density is not None:
         if box is None and n_compounds is not None:
-            total_mass = np.sum([n*np.sum([a.mass for a in c.to_parmed().atoms])
+            if compound_mass is None:
```

Comment:
if we add a mass attribute, I think checking whether the particle masses have been set individually would be good before assuming they are atomistic.